userspace/libsinsp/test/events_file.ut.cpp

```diff
@@ -571,3 +571,1127 @@ TEST_F(sinsp_with_test_input, signalfd4)
+	ASSERT_EQ(get_field_as_string(evt, "fspath.nameraw"), "/tmp/test");
+
+	evt = add_event_advance_ts(increasing_ts(), 1, PPME_SYSCALL_UNLINK_X, 1, failed_res);
+	ASSERT_FALSE(field_exists(evt, "fspath.name"));
+	ASSERT_FALSE(field_exists(evt, "fspath.nameraw"));
+}
+
+TEST_F(sinsp_with_test_input, fspath_field_unlinkat)
+{
+	add_default_init_thread();
+	open_inspector();
+	sinsp_evt* evt = NULL;
+	const char *path = "/tmp/test";
+	int dirfd = 0;
+	int64_t res = 0;
+	int64_t failed_res = -1;
+	evt = add_event_advance_ts(increasing_ts(), 1, PPME_SYSCALL_UNLINKAT_E, 2, dirfd, path);
+	ASSERT_FALSE(field_exists(evt, "fspath.name"));
+	ASSERT_FALSE(field_exists(evt, "fspath.nameraw"));
+
+	evt = add_event_advance_ts(increasing_ts(), 1, PPME_SYSCALL_UNLINKAT_X, 1, res);
+	ASSERT_EQ(get_field_as_string(evt, "fspath.name"), "/tmp/test");
+	ASSERT_EQ(get_field_as_string(evt, "fspath.nameraw"), "/tmp/test");
+
+	evt = add_event_advance_ts(increasing_ts(), 1, PPME_SYSCALL_UNLINKAT_X, 1, failed_res);
+	ASSERT_FALSE(field_exists(evt, "fspath.name"));
+	ASSERT_FALSE(field_exists(evt, "fspath.nameraw"));
+}
+
+TEST_F(sinsp_with_test_input, fspath_field_unlink_2)
+{
+	add_default_init_thread();
+	open_inspector();
+	sinsp_evt* evt = NULL;
+	const char *path = "/tmp/test";
+	int64_t res = 0;
+	int64_t failed_res = -1;
+	evt = add_event_advance_ts(increasing_ts(), 1, PPME_SYSCALL_UNLINK_2_E, 0);
+	ASSERT_FALSE(field_exists(evt, "fspath.name"));
+	ASSERT_FALSE(field_exists(evt, "fspath.nameraw"));
+
+	evt = add_event_advance_ts(increasing_ts(), 1, PPME_SYSCALL_UNLINK_2_X, 2, res, path);
+	ASSERT_EQ(get_field_as_string(evt, "fspath.name"), "/tmp/test");
+	ASSERT_EQ(get_field_as_string(evt, "fspath.nameraw"), "/tmp/test");
+
+	evt = add_event_advance_ts(increasing_ts(), 1, PPME_SYSCALL_UNLINK_2_X, 2, failed_res, path);
+	ASSERT_FALSE(field_exists(evt, "fspath.name"));
+	ASSERT_FALSE(field_exists(evt, "fspath.nameraw"));
+}
+
+TEST_F(sinsp_with_test_input, fspath_field_unlinkat_2)
+{
+	add_default_init_thread();
+	open_inspector();
+	sinsp_evt* evt = NULL;
+	const char *path = "/tmp/test";
+	int dirfd = 0;
+	int64_t res = 0;
+	int64_t failed_res = -1;
+	int32_t flags = 0;
+	evt = add_event_advance_ts(increasing_ts(), 1, PPME_SYSCALL_UNLINKAT_2_E, 0);
+	ASSERT_FALSE(field_exists(evt, "fspath.name"));
+	ASSERT_FALSE(field_exists(evt, "fspath.nameraw"));
+
+	evt = add_event_advance_ts(increasing_ts(), 1, PPME_SYSCALL_UNLINKAT_2_X, 4, res, dirfd, path, flags);
+	ASSERT_EQ(get_field_as_string(evt, "fspath.name"), "/tmp/test");
+	ASSERT_EQ(get_field_as_string(evt, "fspath.nameraw"), "/tmp/test");
+
+	evt = add_event_advance_ts(increasing_ts(), 1, PPME_SYSCALL_UNLINKAT_2_X, 4, failed_res, dirfd, path, flags);
+	ASSERT_FALSE(field_exists(evt, "fspath.name"));
+	ASSERT_FALSE(field_exists(evt, "fspath.nameraw"));
+}
+
+TEST_F(sinsp_with_test_input, fspath_field_open)
+{
+	add_default_init_thread();
+	open_inspector();
+	sinsp_evt* evt = NULL;
+	const char *path = "/tmp/test";
```

Comment:
Can we test with a more complicated test paths like `../\.../../tmp/test` or even some more edge case strategically sprinkled across the tests to more accurately test name and nameraw like also once just the file name without a path etc?

Reply:
Ok added test cases for the complex example `../\.../../tmp/test` as well as a bare name without any path.